pkg/hubble/observer/local_observer.go

```diff
@@ -295,7 +296,9 @@ func (s *LocalObserverServer) GetFlows(
 		return err
 	}
 
-	mask, err := fieldmask.New(req.Experimental.GetFieldMask())
+	// TODO: Remove req.Experimental.GetFieldMask after v1.17
+	fm := fieldmaskpb.Union(req.GetFieldMask(), req.Experimental.GetFieldMask())
```

Comment:
I think instead of the union it may be less error prone to choose one over the other given the presence of the top level?

Reply:
If we wanted to go down that route, I think it would be best to error if both are set instead, rather then (somewhat arbitrarily) picking one over the other. WDYT?

Reply:
I think what you're suggesting could also work, but is a bit less flexible.

From the client perspective, I think there are benefits of selecting top level over experimental. That allows a client to communicate with multiple backend versions in the same manner without encountering an error, but perhaps that is over-complicating the situation.

Reply:
If that's the goal, then the union approach I've written would let you communicate with multiple backend versions without errors too. 

I just dislike the idea of having one field take precedence over the other, it's not very intuitive or easy to handle from a client perspective.

Reply:
The problem with the union is if the outer field sets `bar=1` and inner `bar=2` which one takes precedence? Or they could both set completely different fields.

I think it's better to deterministically always choose one over the other, or error out.

Reply:
With union, it'll include all paths from both field masks. So if you have`fieldMasks: ["ip"]` and `experiments.fieldMasks: ["ip.source"]` then you would get all of `ip`. So you'll get the 'biggest' projection when unioned, which seems correct. I think this is quite reasonable. There is still no precedence, we're effectively working with a set (of field names) and set unions don't have a precedence since the result is the same regardless of the order they're unioned.

Reply:
I struggle to see why union is relevant here. The most likely scenario is for the client to be sending the same mask in both outer and inner field for backwards compatibility reasons until a version or two from now. 

The backend should just prefer the new location from the other if it's at the version where proto is current enough to decode both.

We don't need to overcomplicate here

Reply:
Since we don't seem to agree on semantics, can we maybe just defer to performance and complexity?

For me, it's easier to explain what happens with a simple if-else and also no need to go down the codepath to compare the two to save some cpu.

Reply:
Updated